pkg/imgpkg/cmd/copy.go

```diff
@@ -309,76 +310,150 @@ func (c *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
 	return bundleLock.WriteToPath(c.LockOutputFlags.LockFilePath)
 }
 
-func processedImagesMediaType(processedImages *ctlimgset.ProcessedImages) []string {
-	everyMediaType := []string{}
+type nonDistributableLayers struct {
+	imgRef     string
+	imgFullRef string
+	layers     []string
+}
+
+func getNonDistributableLayersFromImageDescriptors(ids *imagedesc.ImageRefDescriptors) []nonDistributableLayers {
```

Comment:
it would be great to move these functions into its own file (since they are self-contained and mostly misc) so that they do not "pollute" main copy file.

Reply:
done